pkg/transform/oauth/github.go

```diff
@@ -12,18 +12,27 @@ import (
 //IdentityProviderGitHub is a Github specific identity provider
 type IdentityProviderGitHub struct {
 	identityProviderCommon `yaml:",inline"`
-	GitHub                 struct {
-		HostName string `yaml:"hostname"`
-		CA       struct {
-			Name string `yaml:"name"`
-		} `yaml:"ca"`
-		ClientID     string `yaml:"clientID"`
-		ClientSecret struct {
-			Name string `yaml:"name"`
-		} `yaml:"clientSecret"`
-		Organizations []string `yaml:"organizations"`
-		Teams         []string `yaml:"teams"`
-	} `yaml:"github"`
+	GitHub                 GitHub `yaml:"github"`
+}
+
+// GitHub provider specific data
+type GitHub struct {
+	HostName      string             `yaml:"hostname"`
+	CA            GitHubCA           `yaml:"ca"`
+	ClientID      string             `yaml:"clientID"`
+	ClientSecret  GitHubClientSecret `yaml:"clientSecret"`
+	Organizations []string           `yaml:"organizations"`
+	Teams         []string           `yaml:"teams"`
+}
```

Comment:
If you do what I stated regarding the CA and ClientSecret your github.go file turns into a single `GitHub` struct that looks like this:
```golang
type GitHub struct {
	HostName      string             `yaml:"hostname"`
	CA            CA                 `yaml:"ca"` // OR CA CertificateAuthority `yaml:"ca"`
	ClientID      string             `yaml:"clientID"`
	ClientSecret  ClientSecret `yaml:"clientSecret"`
	Organizations []string           `yaml:"organizations"`
	Teams         []string           `yaml:"teams"`
}
```